js/tree.js

```diff
@@ -325,32 +332,156 @@
+						tree: self.$element,
+						reportedClosed: reportedClosed
+					});
+					self.$element.off('loaded.fu.tree', self.$element, closedReported);
+				}
+			};
+
+			//trigger callback when all folders have reported opened
+			self.$element.on('closed.fu.tree', closedReported);
+
+			self.$element.find(".tree-branch.tree-open:not('.hide')").each(function () {
+				self.closeFolder(this);
+			});
+		},
+
+		//disclose visible will only disclose visible tree folders
+		discloseVisible: function discloseVisible() {
+			var self = this;
+			var $openableFolders = self.$element.find(".tree-branch:not('.tree-open, .hide')");
+			var reportedOpened = [];
+
+			var openReported = function openReported(event, opened) {
+				reportedOpened.push(opened);
+
+				if (reportedOpened.length === $openableFolders.length) {
+					self.$element.trigger('disclosedVisible.fu.tree', {
+						tree: self.$element,
+						reportedOpened: reportedOpened
+					});
+					/*
+					* Unbind the `openReported` event. `discloseAll` may be running and we want to reset this
+					* method for the next iteration.
+					*/
+					self.$element.off('loaded.fu.tree', self.$element, openReported);
 				}
+			};
+
+			//trigger callback when all folders have reported opened
+			self.$element.on('loaded.fu.tree', openReported);
+
+			// open all visible folders
+			self.$element.find(".tree-branch:not('.tree-open, .hide')").each(function triggerOpen() {
+				self.openFolder($(this).find('.tree-branch-header'), true);
 			});
+		},
+
+		/**
+		* Disclose all will keep listening for `loaded.fu.tree` and if `$(tree-el).data('keep-disclosing')`
+		* is `true` (defaults to `true`) it will attempt to disclose any new closed folders than were
+		* loaded in during the last disclosure.
+		*/
+		discloseAll: function discloseAll() {
+			var self = this;
+
+			//first time
+			if (typeof self.$element.data('disclosures') === 'undefined') {
+				self.$element.data('disclosures', 0);
+				self.$element.data('keep-disclosing', true);
+			}
+
+			//stop the recursion if they throw the kill switch, or if there isn't anything left to do
+			if (self.$element.data('keep-disclosing') && self.$element.find(".tree-branch:not('.tree-open, .hide')").length !== 0) {
+				/*
+				* give the dev a hook to listen for exceeding disclusure limit `exceededDisclosuresLimit`.
+				* We can't do `disclosures >= upperLimit` because then if they call it again it will
+				* exit immediately unless they "reset the machine". If instead we check to see if
+				* `disclosures` is a multiple of `upperLimit`, it will run exactly `upperLimit` times
+				* while preserving (as an accurate data point for the dev) the number of `disclosures`
+				* that have occurred on this element since page load
+				*/
+				self.$element.data('disclosures', self.$element.data('disclosures') + 1);
+				if (self.options.disclosuresUpperLimit >= 1 && (self.$element.data('disclosures') % self.options.disclosuresUpperLimit) === 0) {
```

Comment:
Because I added the later today without fully thinking through the implications ;)

My original thought process doesn't really matter, sufficed to say, that I've simplified the whole thing by trashing the `%` one.